presto-main/src/main/java/io/prestosql/SessionRepresentation.java

```diff
@@ -270,12 +271,17 @@ public ResourceEstimates getResourceEstimates()
     }
 
     public Session toSession(SessionPropertyManager sessionPropertyManager)
+    {
+        return toSession(sessionPropertyManager, emptyMap());
+    }
+
+    public Session toSession(SessionPropertyManager sessionPropertyManager, Map<String, String> extraCredentials)
```

Comment:
I intentionally avoid having `extraCredentials` in `SessionRepresentation` object because it is a public object that exposed in REST endpoints. We need to make sure the credentials are securely stored and transferred within Presto.

Reply:
That makes sense. Let me talk to @dain and see if he has other ideas on how this could be handled. For now, let's put this directly after `session` in `TaskUpdateRequest` since it's basically part of session, and add a comment explaining why it is separate.

Reply:
`SessionRepresentation` is sent to browsers everytime we load the UI, and we definitely don't want to send credentials there (even if the UI was authenticated and used TLS).